test/built-ins/Set/set-iterator-close-after-add-failure.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 23.2.1.1
+description: >
+    Set ( [ iterable ] )
+
+    When the Set function is called with optional argument iterable the following steps are taken:
+
+    ...
+    9. Repeat
+      a. Let next be IteratorStep(iter).
+      b. ReturnIfAbrupt(next).
+      c. If next is false, return set.
+      d. Let nextValue be IteratorValue(next).
+      e. ReturnIfAbrupt(nextValue).
+      f. Let status be Call(adder, set, «nextValue.[[value]]»).
+      g. If status is an abrupt completion, return IteratorClose(iter, status).
+
+---*/
+
+var count = 0;
+var iterable = {};
+iterable[Symbol.iterator] = function() {
+  return {
+    next: function() {
+      return { value: null, done: false };
+    },
+    'return': function() {
```

Comment:
this is Sparta^d^d^d^d^d^dES6 and we do not have to limit ourselves to ES3 silliness.